src/main/java/com/boot/cart/service/CartService.java

```diff
@@ -16,236 +22,256 @@
+            productDTO = productServiceClient.callGetProductByProductName(productName);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Product: " + productName + " not found in the Database!");
         }
 
-        if (product.getProductStock() == 0) {
+        if (productDTO.getStock() == 0) {
             throw new InvalidInputDataException("We are sorry, but currently: " + productName + " is out of order!");
         }
 
-        if (product.getProductStock() < quantity) {
-            throw new InvalidInputDataException("You can not add more than: " + product.getProductStock() + " "
+        if (productDTO.getStock() < quantity) {
+            throw new InvalidInputDataException("You can not add more than: " + productDTO.getStock() + " "
                     + productName + " Products to your shopping cart!");
         }
-        User user;
+        UserDTO user;
         try {
-            user = UserMapper.DtoToUserEntity(userServiceClient.callGetUserByEmail(email));
+            user = userServiceClient.callGetUserByEmail(email);
         } catch (HttpClientErrorException.NotFound e) {
             throw new EntityNotFoundException("Email: " + email + " not found in the Database!");
         }
-        Cart cart = cartRepository.findByUser(user);
+        Cart cart = cartRepository.findByUserId(user.getId());
 
-        List<Product> productList;
+        List<CartEntry> cartEntries;
         double productTotal = 0;
+
         if (cart == null) {
             cart = new Cart();
-            productList = new ArrayList<>();
+            cartEntries = new ArrayList<>();
+            CartEntry cartEntry = new CartEntry();
+            cartEntry.setProductName(productName);
+            cartEntry.setQuantity(quantity);
+            cartEntry.setCart(cart);
+
+            cartEntries.add(cartEntry);
         } else {
-            productList = cart.getProductList();
+            cartEntries = cart.getEntries();
+            for (CartEntry cartEntry : cartEntries) {
+                if (cartEntry.getProductName().equals(productName)) {
+                    cartEntry.setQuantity(cartEntry.getQuantity() + quantity);
+                    cartEntry.setCart(cart);
+                    cartEntryRepository.save(cartEntry);
+                    break;
+                }
+            }
+            List<String> productNames =
+                    cartEntries.stream()
+                            .map(CartEntry::getProductName)
+                            .collect(Collectors.toList());
+
+           if(!productNames.contains(productName)){
+                CartEntry cartEntry1 = new CartEntry();
+                cartEntry1.setProductName(productName);
+                cartEntry1.setQuantity(quantity);
+                cartEntry1.setCart(cart);
+                cartEntryRepository.save(cartEntry1);
```

Comment:
if you add CascadeType.ALL on the cartEntries relation from the Cart, you need to invoke only cartRepository.save and it will persist/merge also the cart entries.

Reply:
done!

Reply:
![Capture6](https://user-images.githubusercontent.com/59165649/171792968-5183c1a0-fc65-494f-851f-438ced4ac63f.PNG)
new CartEntry wount persist if I dont save it.

Reply:
add it to the cartEntries list and it should work.